fs/vfs/fs_epoll.c

```diff
@@ -180,10 +181,11 @@ static int epoll_do_create(int size, int flags)
   if (eph == NULL)
     {
       set_errno(ENOMEM);
-      return -1;
+      return ERROR;
     }
 
-  nxsem_init(&eph->sem, 0, 0);
```

Comment:
The new behavior is a bit different from the original one. I do not think that should matter, but in "old" code the semaphore kept locked after init and only unlocked before return. Do we need to take lock and hold until `file_allocate` return `OK`?

Reply:
It isn't a problem since the user can't do anything before we return fd from epoll_do_open.